client/extensions/wp-job-manager/components/settings/pages/page-dropdown.jsx

```diff
@@ -0,0 +1,28 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import ReduxFormSelect from 'components/redux-forms/redux-form-select';
+
+const PageDropdown = ( { name, pages, ...otherProps } ) => {
+	return (
+		<ReduxFormSelect name={ name } { ...otherProps }>
+			{ pages.map( ( { ID, title } ) => (
+				<option key={ ID } value={ ID }>
+					{ title }
+				</option>
+			) ) }
+		</ReduxFormSelect>
+	);
+};
```

Comment:
Would it make sense to `connect()` this component to the relevant pages selectors (and possibly move `QueryPosts` here, too)?